### PR TITLE
Add some more immutable .NET types

### PR DESCRIPTION
**Pull request by j3parker**
This drops 51

### CODE REVIEW

File: src/D2L.CodeStyle.Analyzers/Common/MutabilityInspector.cs [hunk 1]
```diff
@@ -24,6 +24,7 @@ public sealed class MutabilityInspector {
 			"System.DateTime",
 			"System.Guid",
 			"System.Lazy",
+			"System.Net.IPAddress",
```
Comment (saquibmian) [inline src/D2L.CodeStyle.Analyzers/Common/MutabilityInspector.cs]: Definitely mutable: https://referencesource.microsoft.com/#System/net/System/Net/IPAddress.cs,341

Comment (j3parker) [inline src/D2L.CodeStyle.Analyzers/Common/MutabilityInspector.cs]: Ah I missed the obsolete method on MSDN :(

Comment (saquibmian) [inline src/D2L.CodeStyle.Analyzers/Common/MutabilityInspector.cs]: There's also this guy, but don't know what `Scope` is: https://referencesource.microsoft.com/#System/net/System/Net/IPAddress.cs,412

Comment (j3parker) [inline src/D2L.CodeStyle.Analyzers/Common/MutabilityInspector.cs]: Yep super broken, my bad

**Comment by j3parker** [comment]
This fellas still good for 46

**Comment by saquibmian** [comment]
`System.Workflow.ComponentModel.DependencyProperty` is technically mutable (non-readonly), but not through the public interface, so I'm cool with it.

:shipit: (after you've bumped the version number)

**Comment by saquibmian** [comment]
fyi: if you want the counter to pick it up, you'll have to release a new version of it too.

**Comment by j3parker** [comment]
yep

**Pull request merged**